lib/onelogin/ruby-saml/utils.rb

```diff
@@ -15,7 +15,21 @@ class Utils
 
       DSIG      = "http://www.w3.org/2000/09/xmldsig#"
       XENC      = "http://www.w3.org/2001/04/xmlenc#"
-      DURATION_FORMAT = %r(^(-?)P(?:(?:(?:(\d+)Y)?(?:(\d+)M)?(?:(\d+)D)?(?:T(?:(\d+)H)?(?:(\d+)M)?(?:(\d+)S)?)?)|(?:(\d+)W))$)
+      DURATION_FORMAT = %r(^
+        (-?)P                         # 1: Duration sign
+        (?:
+          (?:(\d+)Y)?                 # 2: Years
+          (?:(\d+)M)?                 # 3: Months
+          (?:(\d+)D)?                 # 4: Days
```

Comment:
Shouldn't weeks (W) go before days (D)?

Reply:
From [the spec](https://www.loc.gov/standards/datetime/iso-tc154-wg5_n0038_iso_wd_8601-1_2016-02-16.pdf), 4.4.3.2:

> In	both	basic and extended format the complete representation of the expression for duration shall be [PnnW] or [PnnYnnMnnDTnnHnnMnnS].

So I think the current week-handling is correct.

Reply:
Hmmm... you are correct regarding the spec, however following the [https://en.wikipedia.org/wiki/Robustness_principle], since this a reading method we should still add it inline, as there doesn't seem to be any harm in doing so (weeks and years/months/days/etc. can coexist without conflict in the same string--just add them together).

Reply:
Since this is one of the few areas where the spec is mercifully clear, and the code prior to this PR already reflected the spec, I'm not keen to change it. We can always open another PR if there's a genuine need to support invalid duration strings, but I'd like to keep this one focused.

Reply:
> ...programs that receive messages should accept non-conformant input as long as the meaning is clear.

The meaning is 100% clear here if weeks are given alongside years, months, and days.

Reply:
To repeat: I'd like to focus this PR on ensuring that we accept as many ISO8601-valid durations as possible. If you'd like to pursue support for invalid durations, by all means open another PR.

Reply:
Fair enough. Let's get this one merged then I'll open another one :)